pkg/network/ovn_kubernetes.go

```diff
@@ -1455,11 +1451,25 @@ func shouldUpdateOVNKonPrepull(ovn bootstrap.OVNBootstrapResult, releaseVersion
 	return true, false
 }
 
-// isUserIPsecMachineConfigPresent returns true if user owned MachineConfigs for IPsec plugin
-// are already present either in master or worker nodes, otherwise returns false.
-func isUserIPsecMachineConfigPresent(infra bootstrap.InfraStatus) bool {
-	return (infra.MasterIPsecMachineConfig != nil && !containsNetworkOwnerRef(infra.MasterIPsecMachineConfig.OwnerReferences)) ||
-		(infra.WorkerIPsecMachineConfig != nil && !containsNetworkOwnerRef(infra.WorkerIPsecMachineConfig.OwnerReferences))
+func isCNOIPsecMachineConfigExists(infra bootstrap.InfraStatus) bool {
+	var exists bool
+	for _, machineConfig := range infra.MasterIPsecMachineConfigs {
+		if containsNetworkOwnerRef(machineConfig.OwnerReferences) {
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		return false
+	}
+	for _, machineConfig := range infra.WorkerIPsecMachineConfigs {
+		if containsNetworkOwnerRef(machineConfig.OwnerReferences) {
+			exists = true
+			break
+		}
+		exists = false
+	}
+	return exists
```

Comment:
you can just `return  true` here and just `return false` at the end

Or

```
func isCNOIPsecMachineConfigPresent(infra bootstrap.InfraStatus) bool {
        isCNOIPsecMachineConfigPresentIn := func(mcs [] infra.MasterIPsecMachineConfigs) bool {
            for _, mc := range mcs {
		    if containsNetworkOwnerRef(mc.OwnerReferences) {
			    return true
		    }
	    }
	    return false
        }
        return isCNOIPsecMachineConfigPresentIn(infra.MasterIPsecMachineConfigs) && isCNOIPsecMachineConfigPresentIn(infra.WorkerIPsecMachineConfigs)
}
```

Reply:
usage of closure `isCNOIPsecMachineConfigPresentIn` looks good. changed it.